Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -895,6 +895,7 @@ internal struct BinaryDecoder: Decoder {
 
         if let k = k, let v = v {
             value[k] = v
+            consumed = true
```

Comment:
Ok, found some old email to the protobuf-team confirming the default value part.  I'll look at fixing that up in another PR.